Assets/Scripts/Guns/EdgeGun.cs

```diff
@@ -29,46 +61,68 @@ public void OnMoveDown(Transform playerTransform, Camera camera)
           out var hitInfo,
           hitDistance);
 
-      if (!hit) 
-        return;
-      
+      if (!hit)
+        return ;
+
       var gameObjectHit = hitInfo.collider.gameObject;
       var currentlyHitNode = graphService.FindNodeByGameObject(gameObjectHit);
 
       // If not a node, don't do anything
-      if (currentlyHitNode == null) 
+      if (currentlyHitNode == null)
         return;
 
       // If player hit the same node twice, don't do anything
-      if (currentlyHitNode == previouslyHitNode) 
+      if (currentlyHitNode == previouslyHitNode)
+      {
+        previouslyHitNode.gameObject.DisableGlow();
+        previouslyHitNode = null;
         return;
-
+      }
 
       if (previouslyHitNode == null)
       {
         previouslyHitNode = currentlyHitNode;
         previouslyHitNode.gameObject.EnableGlow();
-        previouslyHitNode.gameObject.SetGlow(nodeHitGlowStrength);
+        var glowColor = Mode == EdgeMode.Create ? _createGlowColor : _deleteGlowColor;
+        previouslyHitNode.gameObject.SetGlow(glowColor * nodeHitGlowStrength);
         return;
       }
 
-      //Don't create duplicate edge
-      if (graphService.FindEdgeByNodes(previouslyHitNode, currentlyHitNode) != null) 
-        return;
+      var existingEdge = graphService.FindEdgeByNodes(previouslyHitNode, currentlyHitNode);
+
+      switch (Mode)
+      {
+        case EdgeMode.Create:
+          {
+            //Don't create duplicate edge
+            if (existingEdge != null)
```

Comment:
this if is duplicated in both cases. Move to before switch

Reply:
it's not duplicated 
create -> !=
delete -> ==